src/transformers/models/gemma/modeling_gemma.py

```diff
@@ -1082,7 +1085,8 @@ def forward(
 
         hidden_states = outputs[0]
         logits = self.lm_head(hidden_states)
-
+        logits = logits.float()
```

Comment:
I'm not sure why in the first place there is this cast to float in other archs.

Reply:
Logits must be cast to float32 since doing softmax on float16 will cause numerical errors - hence the .float(). I'm not sure if torch autocast auto casts it to float32, most likely not